Add support for detecting release version on CentOS 5 and CentOS 6

irasnyd opened the pull request:
This is implemented by parsing the /etc/redhat-release file, looking for
the first instance of any two numbers separated by a '.' character. This
has held true for all versions of CentOS 5 and 6.

Examples of /etc/redhat-release from both versions:
CentOS release 5.11 (Final)
CentOS release 6.7 (Final)

trustin commented:
@irasnyd Sorry for a late reply. Could your modification also be applied for RHEL and Fedora? If so, could you update your PR so that it also detects RHEL, Fedora, and potentially other distributions?

irasnyd commented:
From my experience with CentOS, just matching the first number in `/etc/redhat-release` is sufficient to retrieve the distro major version number. The URL below shows that this will work for Fedora also. (I use CentOS exclusively, so I cannot test on RHEL or Fedora.)

http://www.putorius.net/2012/09/how-to-find-which-version-of-red-hat-is.html

I moved the version detection outside of the if statement, as it now applies to all of the older distros which use the `/etc/redhat-release` file. I also added a new regex which just matches a single number, since the `VERSION_REGEX` matches two numbers separated by a dot. This is so that Fedora will work (Fedora does not have sub-versions).

trustin commented:
Thanks for a quick update! LGTM.

@nmittler, wanna take a look as well?

nmittler commented:
LGTM ... thanks @irasnyd :)
